source/common/event/file_event_impl.cc

```diff
@@ -32,7 +32,8 @@ void FileEventImpl::assignEvents(uint32_t events) {
   event_assign(&raw_event_, base_, fd_,
                EV_PERSIST | (trigger_ == FileTriggerType::Level ? 0 : EV_ET) |
                    (events & FileReadyType::Read ? EV_READ : 0) |
-                   (events & FileReadyType::Write ? EV_WRITE : 0),
+                   (events & FileReadyType::Write ? EV_WRITE : 0) |
```

Comment:
Do you want to add the ::Closed case to ::activate above? Or at least add an ASSERT (and doc) that it's not supported?